### ISSUE TITLE
[yeelight] Yeelight Color Bulb Gen 1 - Switch and Brigthness needed

### ISSUE THREAD

**Issue by Selters** [openhab/openhab2-addons#4607]
## Expected Behavior
How can one switch this bulb ("wonder") on/off and adjust the brightness?

## Current Behavior
There is no switch to switch the bulb on/off

## Possible Solution
With the color picker item it is possible to switch the bulb.
“Arrow up” turns on the light and “arrow down” turns it off.

## Steps to Reproduce (for Bugs)

When switching off the “wonder” bulb via yeelight app I got the following output:
`[INFO ] [light.internal.lib.device.DeviceBase] - status = DeviceStatus [isPowerOff=true, r=255, g=0, b=0, color=16711680, brightness=100, ct=3908, hue=359, sat=100, isFlowing=false, delayOff=-1, mFlowItems=null, mode=MODE_SUNHINE, isMusicOn=false, name=]`

## Context

This [workaround ](https://community.openhab.org/t/yeelight-binding/13735/177) doesn't work

## Your Environment
* Version used: oH2.4 with yeelight binding from market
* Environment name and version (e.g. Chrome 39, node.js 5.4, Java 8, ...): Openhabian
* Operating System and version (desktop or mobile, Windows 10, Raspbian Jessie, ...): Openhabian
* It's the [first generation](https://i.ebayimg.com/images/g/pQoAAOSwWflaOX7Z/s-l1600.jpg) of the yeelight color bulb

**Comment by claell** [comment]
The "workaround" you linked is actually how it is meant to be used. To me it seems to be rather that switching on and off (plus brightness -> see below) does not work for you.

Just for clarification, brightness is also not working for you (using the slider from the color picker, maybe with saturation and hue set to not zero)?

**Comment by Selters** [comment]
According to the doc, there are only color and colorTemperature channels for wonder bulb.

![image 107](https://user-images.githubusercontent.com/46726904/51274579-d4c33b80-19cf-11e9-8ecf-b28f599cd32a.png)

So, brightness is missing - which - at least - is needed to swich the bulb on and off. 
But an extra switch channel would be great too ...

**Comment by sverzijl** [comment]
> Why? It might be a bit easier to use when not knowing that the same can already be done with the color channel. 

I've come across this issue too. 

The problem is that other apps (like the official one or Google Home) do not use brightness to turn off the light. The issue then is that there is no way to tell if the light is on or off when it's been controlled elsewhere.

**Comment by claell** [comment]
Hm, that might be a point. I am wondering though, whether this can be still achieved with the current configuration. Maybe the binding has to be changed to make it working with the color channel.

At least the LIFX binding does it like that. So either it has the same problem meaning the concept of only having color and color temperature channel has to be overthought or it handles it in a different way that can then be implemented for the Yeelight binding as well.

**Comment by sverzijl** [comment]
So when Google Home, for example, turns off the light the binding returns this:
`2019-01-29 22:15:25.797 [INFO ] [light.internal.lib.device.DeviceBase] - status = **DeviceStatus [isPowerOff=true**, r=255, g=127, b=0, color=16744192, brightness=50, ct=4000, hue=359, sat=100, isFlowing=false, delayOff=-1, mFlowItems=null, mode=MODE_COLOR, isMusicOn=false, name=]`

The other positive side-effect is when Google Home turns back on the device all the colours and brightness return to their values prior to the light turning off - something that I don't believe can be done in the current set up.

I spent the last hour reading the source code however my knowledge of Java goes as far as its similarities to Python and I'm not sure how everything connects with openhab itself (any information on that last point would be great). I can see that the binding is aware of the status update (at least to produce that log entry). Is it just a matter of making another channel that supports the 'ispoweroff' flag which produces an OnOff result?

**Comment by claell** [comment]
Ok, so at least the binding gets the information.

The question is, whether another channel is really needed or whether there can simply be a switch item linked to the color channel that can then be turned off. At least for turning on and off from inside openHAB that works and afaik it also remembers brightness and color settings, so that might also work in the reverse direction.

If you want to learn more about how bindings work, I think a good starting point is the openHAB documentation pages where they have a section about developing bindings.

**Comment by sverzijl** [comment]
Can you use only the colour channel but have it register the 'ispoweroff'
flag?

My issue is that I have no way to indicate that the light is off if it has
been turn off by something like Google Home.

**Comment by claell** [comment]
Thanks for taking the time to dig in the source code. Interesting to hear that it seemed to be intended to forward the status, but it just is not done properly. That might indeed be some bug (I think these parts you quoted are from a former external library which got now integrated as code directly into the binding. This library is not that nicely written and might contain some bugs).

Feel free to post a separate issue for that to further track this!

**Comment by claell** [comment]
Just for the record, the new issue for that: #4767

**Comment by goodfore** [comment]
@Selters and @claell can we close this issue? I submitted a PR to close #4767

**Comment by claell** [comment]
This resulted from https://community.openhab.org/t/yeelight-binding/13735/233?u=clel which was a different issue. @sverzijl only commented with his issue which was not related, so he created another one for it.

However, I think I found the problem of @Selters now. If I am right it has nothing to do with the binding, but he did not create the correct items.

He tried `Switch	Color_Toggle	 (Gyeelight,Glicht)	{ channel="yeelight:wonder:0x000000000xxxxx:colorTemperature" }` which obviously did not work, since it was the colorTemperature, not the color channel. I did not notice that back then. However I will wait for @Selters to confirm that.
